### DISCUSSION TITLE
sleap keypoints and metadata to .csv or .yaml DLC format

### DISCUSSION THREAD

**Discussion by r-shruthi11** [talmolab/sleap#1089] [Help!] [answered]
Hey y'all,

I'm wondering if you have any utils to go from tracked + proofread .slp files to DLC-type .csv or .yaml formatted files? I see you have [this](https://github.com/talmolab/sleap/blob/develop/sleap/io/convert.py) to go from DLC to sleap but anything for the reverse?

Cheers,
Shruthi

**Comment by roomrys** [comment]
Hi @r-shruthi11,

Unfortunately, we do not currently have an adaptor to go SLEAP to DLC. The only format (other than SLEAP specific formats - .pkg.slp, 
.analysis.h5) that SLEAP currently exports to is [NWB](https://nwb-overview.readthedocs.io/en/latest/tools/tools_home.html#behavior) (with an open [PR](https://github.com/talmolab/sleap/pull/1068) for exporting to [Nix](https://nixio.readthedocs.io/en/latest/)).

Thanks,
Liezl

**Comment by sronilsson** [comment, marked as answer]
@r-shruthi11 there is a nb here by @Toshea111 to get it into slp h5 into CSV.  https://colab.research.google.com/drive/1EpyTKFHVMCqcb9Lj9vjMrriyaG9SvrPO?usp=sharing

**Reply by r-shruthi11** [reply]
this looks great - thanks!

**Reply by Toshea111** [reply]
Dear r-shruthi11,

I have provided an updated version of the notebook here:

https://github.com/Toshea111/sleap/blob/develop/docs/notebooks/Convert_HDF5_to_CSV_updated.ipynb

It fixes some issues with compatibility and illegal slicing arguments in the original version.

Best wishes,

Thomas
